pkg/elfparser/elf.go

```diff
@@ -138,7 +161,30 @@ func (b *bpfSDKClient) LoadBpfFile(path, customizedPinPath string) (map[string]B
 
 	elfLoader := newElfLoader(elfFile, b.mapApi, b.progApi, customizedPinPath)
 
-	bpfLoadedProg, bpfLoadedMaps, err := elfLoader.doLoadELF()
+	bpfLoadedProg, bpfLoadedMaps, err := elfLoader.doLoadELF(BpfCustomData{})
+	if err != nil {
+		return nil, nil, err
+	}
+	return bpfLoadedProg, bpfLoadedMaps, nil
+}
+
+func (b *bpfSDKClient) LoadBpfFileWithCustomData(inputData BpfCustomData) (map[string]BpfData, map[string]ebpf_maps.BpfMap, error) {
+
+	bpfFile, err := os.Open(inputData.FilePath)
+	if err != nil {
+		log.Infof("LoadBpfFile failed to open")
```

Comment:
Should be an error instead right? Also fn name in the log line doesn't match (or) maybe we can just say - file create error or something along those lines